Sources/Plasma/CoreLib/CMakeLists.txt

```diff
@@ -83,6 +75,29 @@ set(CoreLib_HEADERS
     plViewTransform.h
 )
 
+set(BUILD_INFO_FILE "${CMAKE_CURRENT_BINARY_DIR}/hsBuildInfo.inl")
+set(BUILD_INFO_COMMAND ${CMAKE_COMMAND}
+    -DPRODUCT_BRANCH_ID=${PRODUCT_BRANCH_ID}
+    -DPRODUCT_BUILD_ID=${PRODUCT_BUILD_ID}
+    -DPRODUCT_BUILD_TYPE=${PRODUCT_BUILD_TYPE}
+    -DPRODUCT_CORE_NAME=${PRODUCT_CORE_NAME}
+    -DPRODUCT_SHORT_NAME=${PRODUCT_SHORT_NAME}
+    -DPRODUCT_LONG_NAME=${PRODUCT_LONG_NAME}
+    -DPRODUCT_UUID=${PRODUCT_UUID}
+    -DPRODUCT_EMBED_BUILD_INFO=${PRODUCT_EMBED_BUILD_INFO}
+    -DPRODUCT_EMBED_BUILD_TIME=${PRODUCT_EMBED_BUILD_TIME}
+    -P "${CMAKE_MODULE_PATH}/BuildInfo.cmake"
+)
+add_custom_target(pcBuildInfo
+    COMMAND ${BUILD_INFO_COMMAND}
+    BYPRODUCTS "${BUILD_INFO_FILE}"
+)
+list(APPEND CoreLib_HEADERS "${BUILD_INFO_FILE}")
+
+# Always generate the hsBuildInfo.inl at configure time to ensure the file is present-and-accounted
+# for. Otherwise, people like me will be staring at the include going, "WTF mate?"
+execute_process(COMMAND ${BUILD_INFO_COMMAND} OUTPUT_QUIET WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
Eh, I'm not sure this is really necessary...  We already have cases of generated files that don't show up until you build (e.g. the Qt automoc/autouic/autorcc stuff), and this will increase the time to re-run cmake each time.